stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```diff
@@ -0,0 +1,1447 @@
+                                       "method."
+                                       .format(str(method.value)))
+            self._request.method = method.value
+
+        # String Check
+        # If the value was not found in a enum then we attempt to find the enum that contains the same as the
+        # string value provided.
+        # The following checks are made
+        # - Is the method passed a string so that we can match it in the enum
+        # - Does that string exist in the mapped enum list.
+        elif isinstance(method, str) and method in self._common.http_method_enums:
+            self._common.print_verbose("String method found inside the `HTTPMethod` enum, Applying {0} as the active"
+                                       "method."
+                                       .format(str(method)))
+            self._request.method = method
+
+        # If we do not find the value inside the enum then it means that we do not support the method yet
+        else:
+            self._common.print_not_implemented_error("""Unable to find the provided {0} method. Currently the code
+                                                       block only supports the methods provided in the `HTTPMethod`
+                                                       Enum. If you would like to add another method feel free to add it
+                                                       within the `HTTPMethod` Enum or change the supplied {1} to a
+                                                       supported type""".format(str(method), str(method)))
+
+    """
+    * The following functions control the HTTP Endpoint and Endpoint creation
+      For example http://www.example.com and http://www.example.com?test=123 will both be mapped to the correct
+      values within the Request() object
+    """
+
+    # Clear the current state of the HTTP URL applied to the Request() object
+    def clear_url(self):
+        self._common.print_verbose("Clearing the HTTP URL.")
+        self._request.url = None
+
+    # Retrieve the value of the current Endpoint on the Request() object
+    def get_url(self):
+        self._common.print_verbose("Retrieving current HTTP Endpoint")
+        return self._request.url
+
+    """
+    Functionality:
+        The Request() endpoint is set with this function.
+        Some extra functionality is build-into this function to analyze a URL. When something like the following is
+        passed to the function 'http://www.url.com?test=123' the query parameters on this URL will be split out and
+        applied into the correct object within the Request() object
+
+    Input:
+        @url
+            A string object containing the endpoint that should be queried
+    """
+
+    def set_url(self, url):
+        self._common.print_verbose("Attempting to set the active current url to {0}".format(str(url)))
```

Comment:
You know. All these log statements, could be drastically simplified if you'd just get a dict with all the information in it that you need to transform to a requests-lib function. You'd then simply log the dict and be finished.

Reply:
The intent for those verbose prints was to indicate a single object before it was applied. This will allow to see what was the action before a possible error.

The dict object might give a bigger picture but not a precise action before a error occured.

When I complete the following comment "Instead of always keeping the request object in sync at all places, you can also just construct the request object on demand through a method. This is safer and easier to follow." I will include the dict inside the logs to indicate the current state and the attempted action